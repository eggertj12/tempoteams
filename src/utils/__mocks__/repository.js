const repo = jest.genMockFromModule('fs');

const resolves = new Promise((resolve) => {
  resolve([])
})
const rejects = new Promise((resolve, reject) => {
  reject()
})

let promise = resolves

repo.get = () => {
  return promise
}

repo.__setResult = data => {
  promise = new Promise((resolve) => {
    resolve(data)
  })
}

repo.__shouldSucceed = () => {
  promise = resolves
}

repo.__shouldFail = () => {
  promise = rejects
}

module.exports = repo;
