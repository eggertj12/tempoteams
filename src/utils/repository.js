function get(what, id = null) {

  // To be read from configuration (ENV??)
  const urls = {
    'teams': 'http://tempo-test.herokuapp.com/7d1d085e-dbee-4483-aa29-ca033ccae1e4/1/team/',
    'team': `http://tempo-test.herokuapp.com/7d1d085e-dbee-4483-aa29-ca033ccae1e4/1/team/${id}/`,
  }

  const url = urls[what]
  if (!url) {
    return Promise.reject('Invalid items requested')
  }
  return fetch(url)
    .then(function (response) {
      return response.json()
    })
}

export { get }
