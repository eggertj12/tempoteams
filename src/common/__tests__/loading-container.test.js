import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import repository from '../../utils/repository'

import LoadingContainer from '../loading-container'

Enzyme.configure({ adapter: new Adapter() })

jest.mock('../../utils/repository')

describe('LoadingContainer', () => {
  function givenData(data) {
    repository.__setResult(data)
  }

  function givenFailure() {
    repository.__shouldFail()
  }

  beforeEach(function () {
    repository.__shouldSucceed()
  })

  it('renders without crashing when given a single child', () => {
    shallow(<LoadingContainer><p /></LoadingContainer>)
  })

  it('requires a child', () => {
    expect(
      () => shallow(<LoadingContainer />)
    ).toThrow()
  })

  it('only allows a single child', () => {
    expect(
      () => shallow(<LoadingContainer><p /><p /></LoadingContainer>)
    ).toThrow()
  })

  it('renders Loading before promise is resolved', () => {
    const rendered = shallow(<LoadingContainer><p /></LoadingContainer>)

    expect(rendered).toMatchSnapshot()
  })

  it('renders Error on failure', done => {
    givenFailure()

    const rendered = shallow(<LoadingContainer><p /></LoadingContainer>)

    setTimeout(() => {
      expect(rendered).toMatchSnapshot()
      done()
    }, 0)
  })

  it('renders Child on success', done => {
    const rendered = shallow(<LoadingContainer><p /></LoadingContainer>)

    setTimeout(() => {
      expect(rendered).toMatchSnapshot()
      done()
    }, 0)
  })

  it('passes data to Child on success', done => {
    givenData({ 'name': 'Awesome team' })

    const wrapper = shallow(<LoadingContainer><p /></LoadingContainer>)

    setTimeout(() => {
      expect(wrapper.prop('data').name).toEqual('Awesome team')
      done()
    }, 0)
  })
})
