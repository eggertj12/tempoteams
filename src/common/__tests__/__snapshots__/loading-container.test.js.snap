// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingContainer renders Child on success 1`] = `
<p
  data={Array []}
/>
`;

exports[`LoadingContainer renders Error on failure 1`] = `
<span>
  ERROR!
</span>
`;

exports[`LoadingContainer renders Loading before promise is resolved 1`] = `
<span>
  Loading ...
</span>
`;
